apps/wallet/src/ui/app/shared/forms/controls/PasswordInput.tsx

```diff
@@ -0,0 +1,35 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import { EyeOpen16, EyeClose16 } from '@mysten/icons';
+import { forwardRef, useState, type ComponentProps } from 'react';
+import { Input } from './Input';
+import { ButtonOrLink } from '../../utils/ButtonOrLink';
+
+type PasswordInputProps = {
+	name: string;
+} & Omit<ComponentProps<'input'>, 'className' | 'type' | 'name' | 'ref'>;
+
+export const PasswordInput = forwardRef<HTMLInputElement, PasswordInputProps>(
+	({ placeholder, ...props }, forwardedRef) => {
+		const [passwordShown, setPasswordShown] = useState(false);
+		const IconComponent = passwordShown ? EyeOpen16 : EyeClose16;
+
+		return (
+			<div className="flex w-full relative items-center">
+				<Input
+					type={passwordShown ? 'text' : 'password'}
+					placeholder="Password"
+					ref={forwardedRef}
+					{...props}
+				/>
+				<ButtonOrLink
+					className="appearance-none bg-transparent border-none cursor-pointer absolute right-3 text-gray-60 peer-focus:text-steel"
```

Comment:
looks like we could use a `flex items-center` here: 

![image](https://github.com/MystenLabs/sui/assets/122397493/0339e1c9-d349-4bef-afb3-62b00a398d11)

Reply:
good 👁️ , thanks